### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by LeonardG1993**
First PR to Main:
- Fully functioning app
- Need to refactor controllers as they use sessions
- Need to update some unit tests
- Currently using local DB for testing

### CODE REVIEW

**Comment by acarasimon96** [comment]
> * Currently using local DB for testing

I don't think the unit tests use any form of database. Rather, I only used dummy model instances and mocks to write them

File: src/main/java/com/revature/controllers/ReviewController.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import com.revature.annotations.Authorized;
+import com.revature.dtos.ReviewRequest;
+import com.revature.models.Review;
+import com.revature.models.User;
+import com.revature.services.ReviewService;
+
+@RestController
+@RequestMapping("/api/review")
+@CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/ReviewController.java]: Change this to `@CrossOrigin(origins="*", allowedHeaders="*")` to allow frontend in prod environment to reach this controller without throwing a CORS error

File: src/main/java/com/revature/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.revature.annotations.Authorized;
+import com.revature.dtos.UserRequest;
+import com.revature.models.User;
+import com.revature.services.AddressService;
+import com.revature.services.UserService;
+
+@RestController
+@RequestMapping("/api/users")
+@CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/UserController.java]: Change this to `@CrossOrigin(origins="*", allowedHeaders="*") `to allow frontend in prod environment to reach this controller without throwing a CORS error

File: src/main/java/com/revature/controllers/PurchaseController.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.revature.annotations.Authorized;
+import com.revature.dtos.PurchaseRequest;
+import com.revature.models.Purchase;
+import com.revature.models.User;
+import com.revature.services.PurchaseService;
+
+@RestController
+@RequestMapping("api/purchases")
+@CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/PurchaseController.java]: Change this to `@CrossOrigin(origins="*", allowedHeaders="*")` to allow frontend in prod environment to reach this controller without throwing a CORS error

File: src/main/java/com/revature/controllers/ProductController.java [hunk 4]
```diff
@@ -1,50 +1,117 @@
...
+import com.revature.annotations.AuthorizedAdmin;
+import com.revature.dtos.CreateUpdateRequest;
+import com.revature.exceptions.InvalidProductInputException;
+import com.revature.exceptions.InvalidRoleException;
+import com.revature.services.StorageService;
 
 @RestController
-@RequestMapping("/api/product")
+@RequestMapping("/api/product" )
 @CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/ProductController.java]: Change this to `@CrossOrigin(origins="*", allowedHeaders="*")` to allow frontend in prod environment to reach this controller without throwing a CORS error

File: src/main/java/com/revature/controllers/AddressController.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+import com.revature.dtos.AddressRequest;
+import com.revature.models.Address;
+import com.revature.models.User;
+import com.revature.services.AddressService;
+import com.revature.services.UserService;
+
+@RestController
+@RequestMapping("/api/addresses")
+@CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```
Comment (sophiagavrila) [inline src/main/java/com/revature/controllers/AddressController.java]: Change this to `@CrossOrigin(origins="*", allowedHeaders="*")` to allow frontend in prod environment to reach this controller without throwing a CORS error

**Review by sophiagavrila — CHANGES_REQUESTED** [review]

**Comment by sophiagavrila** [comment]
Resolve code smells - do not use `System.out.println(...)` where you could instead use a Logger.

Code smell report here https://sonarcloud.io/project/issues?id=Revature-Sophia-ECommerce-1376_e-commerce-backend&pullRequest=54&resolved=false&types=CODE_SMELL

**Comment by sophiagavrila** [comment]
Add Jacoco Plugin to the `pom.xml` to generate test coverage report. It's a simple add - paste from here and follow instructions outlined here https://docs.sonarcloud.io/enriching/test-coverage/java-test-coverage/#:~:text=Add%20coverage%20in,the%20project%20root.

**Pull request closed without merging**